File: Leetcode/2016 - Maximum Difference Between Increasing Elements/solution.py
def maximumDifference(self, nums: List[int]) -> int:
    curr_min = nums[0]
    max_diff = -1

    for i in range(1, len(nums)):
        if nums[i] > curr_min:
            max_diff = max(max_diff, nums[i] - curr_min)
        else:
            curr_min = nums[i]

    return max_diff

File: Leetcode/2016 - Maximum Difference Between Increasing Elements/solution.js
var maximumDifference = function (nums) {
    let currMin = nums[0];
    let maxDiff = -1;
    for (let i = 1; i < nums.length; i++) {
        if (nums[i] > currMin) {
            maxDiff = Math.max(maxDiff, nums[i] - currMin);
        } else {
            currMin = nums[i];
        }
    }
    return maxDiff;
};